indexing-service/src/main/java/org/apache/druid/indexing/overlord/ZkWorker.java

```diff
@@ -63,6 +67,25 @@ public ZkWorker(Worker worker, PathChildrenCache statusCache, final ObjectMapper
     this.statusCache = statusCache;
     this.cacheConverter = (ChildData input) ->
         JacksonUtils.readValue(jsonMapper, input.getData(), TaskAnnouncement.class);
+    this.taskIdExtractor = createTaskIdExtractor(jsonMapper);
+  }
+
+  static java.util.function.Function<ChildData, String> createTaskIdExtractor(final ObjectMapper jsonMapper)
```

Comment:
can we add some tests for this method to understand the expected behavior? This would also ensure it's less likely to break if new fields are added to the task data.

Reply:
this is covered in ZkWorker, however, some of the negative / problematic cases you bring up in other comments do warrant specific tests for this; so i'll add some tests and resolve any issues